gammapy/data/observations.py

```diff
@@ -73,11 +73,31 @@ def __init__(
         self.edisp = edisp
         self.psf = psf
         self.bkg = bkg
-        self.rad_max = rad_max
+        self._rad_max = rad_max
         self._gti = gti
         self._events = events
         self.obs_filter = obs_filter or ObservationFilter()
 
+
+    @property
+    def rad_max(self):
+        # prevent circular import
+        from gammapy.irf import RadMax2D
+
+        if self._rad_max is not None:
+            return self._rad_max
+
+        # load once to avoid trigger lazy loading it three times
+        aeff = self.aeff
+        if aeff is not None and aeff.is_pointlike:
+            self._rad_max = RadMax2D.from_irf(aeff)
+
```

Comment:
Why is `self._rad_max` created twice? Shouldn't it be returned once created?

Reply:
yes, sorry, there is a return statement missing here.

Reply:
fixed